tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Profiler.scala

```diff
@@ -106,6 +107,20 @@ class Profiler(hadoopConf: Configuration, appArgs: ProfileArgs) extends Logging
     }
   }
 
+  private def errorHandler(error: Throwable, path: EventLogInfo) = {
+    error match {
+      case oom: OutOfMemoryError =>
+        logError(s"OOM error while processing large file: ${path.eventLog.toString}." +
+            s" Increase heap size. Exiting ...", oom)
+        sys.exit(1)
+      case NonFatal(e) =>
+        logWarning(s"Exception occurred processing file: ${path.eventLog.getName}", e)
+      case o =>
```

Comment:
nit: there is no warning in the function but if we every move it back inside a catch will get a warning like #3743. 
```suggestion
      case o: Throwable =>
```

Reply:
Thanks! updated it.